src/sentry/static/sentry/app/components/externalIssues/abstractExternalIssueForm.tsx

```diff
@@ -0,0 +1,244 @@
+import React from 'react';
+import {Modal} from 'react-bootstrap';
+import debounce from 'lodash/debounce';
+import * as queryString from 'query-string';
+
+import AsyncComponent from 'app/components/asyncComponent';
+import {IntegrationIssueConfig, IssueConfigField} from 'app/types';
+import FieldFromConfig from 'app/views/settings/components/forms/fieldFromConfig';
+import Form from 'app/views/settings/components/forms/form';
+import {FieldValue} from 'app/views/settings/components/forms/model';
+import {FormField} from 'app/views/settings/projectAlerts/issueEditor/ruleNode';
+
+export type ExternalIssueAction = 'create' | 'link';
+
+type Props = {
+  action?: ExternalIssueAction;
+} & AsyncComponent['props'];
+
+type State = {
+  integrationDetails?: IntegrationIssueConfig;
+  dynamicFieldValues?: {[key: string]: FieldValue};
+} & AsyncComponent['state'];
+
+const DEBOUNCE_MS = 200;
+
+/**
+ * @abstract
+ */
+export default class AbstractExternalIssueForm<
+  P extends Props = Props,
+  S extends State = State
+> extends AsyncComponent<P, S> {
+  shouldRenderBadRequests = true;
+  action: ExternalIssueAction;
+
+  constructor(props: P, context: any) {
+    super(props, context);
+    this.action = (this.props.action || 'create') as ExternalIssueAction;
```

Comment:
not sure i would recommend doing this since you lose reactivity, maybe a getter on the class? Not sure what pattern react people normally use in this case cc @EvanPurkhiser 

```ts
get action() {
	return this.props.action || 'create';
}
```

Reply:
I agree we should avoid type casting here if possible